lib/grp.gi

```diff
@@ -1586,6 +1586,35 @@ function( G )
     return comp;
 end );
 
+#############################################################################
+##
+#M  Socle( <G> )  . . . . . . . . . . . . . . . . for finite nilpotent groups
+##
+InstallMethod(Socle, "for nilpotent groups",
+              [ IsGroup and IsFinite and IsNilpotentGroup ],
+  function(G)
+    local p, H, prodH;
+
+    prodH := TrivialSubgroup(G);
+    # now socle is product of Omega of the center of its Sylow subgroups
+    for H in SylowSystem(G) do
+      p := PrimePGroup(H);
+      prodH := ClosureSubgroupNC(prodH, Omega(Center(H), p));
+
+      # prodH is central in G, set some properties and attributes accordingly
+      SetIsAbelian(prodH, true);
+      if not HasParent(prodH) then
+        SetParent(prodH, G);
+        SetCentralizerInParent(prodH, G);
+        SetIsNormalInParent(prodH, true);
+      elif IsSubgroup(G, Parent(prodH)) then
+        SetCentralizerInParent(prodH, Parent(prodH));
+        SetIsNormalInParent(prodH, true);
+      fi;
+    od;
```

Comment:
I am not entirely sure, but I think in one of my examples prodH did not have a parent... 

However, in any case the parent might not be G, e.g. if G is a FittingGroup of some bigger group, then prodH would inherit the parent of G (see examples from #398). 

I set these filters for the internal groups so that GAP would know about them. It might be a waste of memory, or the user might use these groups later on for something else. I can move it outside of the loop if that is preferred.